src/SonarScanner.MSBuild.PreProcessor/WebService/SonarCloudWebService.cs

```diff
@@ -19,23 +19,125 @@
+            var empty = new List<SensorCacheEntry>();
+
+            if (string.IsNullOrWhiteSpace(localSettings.ProjectKey))
+            {
+                logger.LogInfo(Resources.MSG_Processing_PullRequest_NoProjectKey);
+                return empty;
+            }
+            if (string.IsNullOrWhiteSpace(localSettings.Organization))
+            {
+                logger.LogInfo(Resources.MSG_Processing_PullRequest_NoOrganization);
+                return empty;
+            }
+            if (!localSettings.TryGetSetting(SonarProperties.PullRequestBase, out var branch))
+            {
+                logger.LogInfo(Resources.MSG_Processing_PullRequest_NoBranch);
+                return empty;
+            }
+            if (!localSettings.TryGetSetting(SonarProperties.SonarUserName, out var token))
+            {
+                logger.LogInfo(Resources.MSG_Processing_PullRequest_NoToken);
+                return empty;
+            }
+
+            var serverSettings = await GetProperties(localSettings.ProjectKey, branch);
+            if (!serverSettings.TryGetValue(SonarProperties.CacheBaseUrl, out var cacheBaseUrl))
+            {
+                logger.LogInfo(Resources.MSG_Processing_PullRequest_NoCacheBaseUrl);
+                return empty;
+            }
+
+            try
+            {
+                logger.LogInfo(Resources.MSG_DownloadingCache, localSettings.ProjectKey, branch);
+                var ephemeralUrl = await GetEphemeralCacheUrl(localSettings.Organization, localSettings.ProjectKey, branch, token, cacheBaseUrl);
+                using var stream = await GetCacheStream(ephemeralUrl);
+                return ParseCacheEntries(stream);
+            }
+            catch (Exception e)
+            {
+                logger.LogWarning(Resources.MSG_IncrementalPRCacheEntryDeserialization, e.Message);
+                return empty;
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                cacheClient.Dispose();
+            }
+
+            base.Dispose(disposing);
+        }
+
+        private async Task<Uri> GetEphemeralCacheUrl(string organization, string projectKey, string branch, string token, string cacheBaseUrl)
+        {
+            var uri = new Uri(WebUtils.CreateUri(cacheBaseUrl), WebUtils.Escape("v1/sensor_cache/prepare_read?organization={0}&project={1}&branch={2}", organization, projectKey, branch));
+            using var request = new HttpRequestMessage(HttpMethod.Get, uri);
+            request.Headers.Add("Authorization", $"Bearer {token}");
+
+            using var response = await cacheClient.SendAsync(request);
+            response.EnsureSuccessStatusCode();
+            var content = await response.Content.ReadAsStringAsync();
+            var deserialized = JsonConvert.DeserializeAnonymousType(content, new { Enabled = false, Url = "placeholder" });
+            return new Uri(deserialized.Url);
```

Comment:
Same with every other error in `DownloadEphemeralUrl` or `GetCacheStream`, it will be propagated one-step above and the exception will be catched and logged via:

```csharp
logger.LogWarning(Resources.MSG_IncrementalPRCacheEntryDeserialization, e.Message);
``` 

Maybe we should change the message from deserialization to something more generic?
The message now reads: 

"an error occurred while deserializing the cache entries! {Exception Mesage}"